### ISSUE TITLE
`test_pdb` broken under WASI

### ISSUE THREAD

**Issue by brettcannon** [python/cpython#114275]
# Bug report

https://github.com/python/cpython/pull/112109 is the culprit.

https://buildbot.python.org/all/#/builders/1046/builds/4033
https://buildbot.python.org/all/#/builders/1124/builds/623

### Linked PRs
* gh-114283
* gh-114309
* gh-114438
* gh-114439

**Comment by sobolevn** [comment]
Hm, I am investigating this 🤔 

First of all, it passes locally on macos platform.

Second, here's the failure:

```
Doctest: test.test_pdb.test_pdb_next_command_for_coroutine ... FAIL
test_pdb_next_command_for_generator (test.test_pdb)
Doctest: test.test_pdb.test_pdb_next_command_for_generator ... ok
test_pdb_next_command_in_generator_for_loop (test.test_pdb)
Doctest: test.test_pdb.test_pdb_next_command_in_generator_for_loop ... ok
test_pdb_next_command_subiterator (test.test_pdb)
Doctest: test.test_pdb.test_pdb_next_command_subiterator ... ok
test_pdb_pp_repr_exc (test.test_pdb)
Doctest: test.test_pdb.test_pdb_pp_repr_exc ... Warning -- Unraisable exception
Exception ignored in: <function BaseEventLoop.__del__ at 0xb453f8>
Traceback (most recent call last):
  File "/Lib/asyncio/base_events.py", line 750, in __del__
    self.close()
  File "/Lib/asyncio/unix_events.py", line 70, in close
    super().close()
  File "/Lib/asyncio/selector_events.py", line 104, in close
    self._close_self_pipe()
  File "/Lib/asyncio/selector_events.py", line 111, in _close_self_pipe
    self._remove_reader(self._ssock.fileno())
AttributeError: '_UnixSelectorEventLoop' object has no attribute '_ssock'
Warning -- Unraisable exception
Exception ignored in: <function BaseEventLoop.__del__ at 0xb453f8>
Traceback (most recent call last):
  File "/Lib/asyncio/base_events.py", line 750, in __del__
    self.close()
  File "/Lib/asyncio/unix_events.py", line 70, in close
    super().close()
  File "/Lib/asyncio/selector_events.py", line 104, in close
    self._close_self_pipe()
  File "/Lib/asyncio/selector_events.py", line 111, in _close_self_pipe
    self._remove_reader(self._ssock.fileno())
AttributeError: '_UnixSelectorEventLoop' object has no attribute '_ssock'
ok
test_pdb_return_command_for_coroutine (test.test_pdb)
Doctest: test.test_pdb.test_pdb_return_command_for_coroutine ... FAIL
```

Third, no direct import / usage of `test_doctest` is there.
Looking further.

**Comment by sobolevn** [comment]
Ok, I found out what is going on:
1. `test_pdb` used to import code from `test_doctest`
2. `test_doctest` has `if not support.has_subprocess_support: skip`
3. So, all `test_pdb` was skipped on wasi, proof: https://buildbot.python.org/all/#/builders/1061/builds/694/steps/11/logs/stdio
4. Now, I removed import from `test_doctest`, which is a good thing
5. It unskipped `test_pdb` on WASI

There are several things we can do:
1. Skip these failing tests on WASI with a TODO comment
2. Skip the whole test file
3. Try to fix it (but it is asyncio related and I am not an expert in this module)

@brettcannon @vstinner what are your thoughts on this?

**Comment by vstinner** [comment]
I suggest to skip again the test which was skipped previously.

**Comment by vstinner** [comment]
Issue fixed by https://github.com/python/cpython/commit/efb81a60f5ce7e192095230a0f7ff9684d6f835a

**Comment by brettcannon** [comment]
I was going to say to skip them. 😁 Thanks for the prompt fix!

**Comment by brettcannon** [comment]
I had to reopen this as the fix didn't get backported to 3.12 and 3.11.

**Comment by vstinner** [comment]
> I had to reopen this as the fix didn't get backported to 3.12 and 3.11.

Alright. I merged the 2 backports, thanks. I close again the issue.
